### PR TITLE
fix(web-components): data table header size

### LINKED ISSUE #17680: [Bug]: Sortable data table with the size prop set to "sm", header row doesn't respect the "sm" size
### Package

@carbon/web-components

### Browser

Chrome

### Package version

2.13.1

### React version

_No response_

### Description

I'm trying to use a sortable data table with the size prop set to "sm". However, it seems using sortable causes the header row to stay larger than other table rows. Current behavior I'm seeing on v 2.13.1
Image

### Reproduction/example

https://stackblitz.com/edit/github-kbd9xw-8msd1g?file=index.html

### Steps to reproduce

1) Open the link
2) View the code in index.html
3) View the output that has a taller header row than the rest of the rows.

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Application/PAL

watsonx Assistant

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by Zoe-Gathercole**
Closes #17680

Removed hard coded min block sizes to allow the header row to adjust height automatically for each table size

#### Changelog

**Removed**

- hard coded min block sizes for the sortable data table header

#### Testing / Reviewing

Open the storybook playground for the sortable data table and change the size of the table to see the header row adjust accordingly.
image

### CODE REVIEW

**Comment by Zoe-Gathercole** [comment]
I have read the DCO document and I hereby sign the DCO.

**Review by ariellalgilmore — APPROVED** [review]
nice! LGTM!

**Comment by Zoe-Gathercole** [comment]
Thank you for the review @ariellalgilmore ! The new push is to fix some merge conflicts with the readme 🙂

File: README.md [hunk 1]
```diff
@@ -313,8 +313,8 @@ check out our [Contributing Guide](/.github/CONTRIBUTING.md) and our
     <td align="center"><a href="https://github.com/ziyadzulfikar"><img src="https://avatars.githubusercontent.com/u/56788667?v=4?s=100" width="100px;" alt=""/><br /><sub><b>Ziyad Bin Sulfi</b></sub></a><br /><a href="https://github.com/carbon-design-system/carbon/commits?author=ziyadzulfikar" title="Code">💻</a></td>
   </tr>
   <tr>
-    <td align="center"><a href="https://irfadkp.github.io/"><img src="https://avatars.githubusercontent.com/u/54243898?v=4?s=100" width="100px;" alt=""/><br /><sub><b>IRFAD KP</b></sub></a><br /><a href="https://github.com/carbon-design-system/carbon/commits?author=irfadkp" title="Code">💻</a></td>
```
Comment (Zoe-Gathercole) [inline README.md:317]: With the recent merge conflict in the README, I noticed this was a duplicate entry - see line 312 😊

**Review by kennylam — APPROVED** [review]
LGTM, thanks!

**Pull request merged**